### PR TITLE
Add details for kubectl.kubernetes.io/restartedAt annotation

### PR DESCRIPTION
**Pull request by Biswajit-Mohapatra2**
fixed: #40067

Adding `"kubectl.kubernetes.io/restartedAt"` annotation to kubernetes documentation with proper description.

Changes at: https://kubernetes.io/docs/reference/labels-annotations-taints/

### CODE REVIEW

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 1]
```diff
@@ -92,6 +92,17 @@ Common forms of values include:
 
 One of the [recommended labels](/docs/concepts/overview/working-with-objects/common-labels/#labels).
 
+### kubectl.kubernetes.io/restartedAt
+
+Example: `kubectl.kubernetes.io/restartedAt: '2023-03-05T15:30:00Z`
+
+Used on: Pod
+
+In this example, the annotation indicates that the Pod was most recently restarted at 3:30 PM UTC on March 31st, 2023.
```
Comment (NitishKumar06) [inline content/en/docs/reference/labels-annotations-taints/_index.md:101]:
```suggestion
In the above example, the annotation indicates that the Pod was most recently restarted at 3:30 PM UTC on March 31, 2023.
```

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 2]
```diff
@@ -92,6 +92,17 @@ Common forms of values include:
...
+
+The `"kubectl.kubernetes.io/restartedAt"` annotation is used by Kubernetes to keep track of the timestamp of the most recent container restart for a particular pod. This information can be useful for debugging and monitoring purposes.
+
+The `"kubectl.kubernetes.io/restartedAt"` annotation is automatically added to the spec template by kubectl when the workload is restarted. It records the time of the most recent workload restart as a Unix timestamp.
```
Comment (NitishKumar06) [inline content/en/docs/reference/labels-annotations-taints/_index.md:105]: This annotation is same as mentioned in line 103. Kindly correct it.

**Review by NitishKumar06 — COMMENTED** [review]
PTAL!

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 3]
```diff
@@ -92,6 +92,18 @@ Common forms of values include:
 
 One of the [recommended labels](/docs/concepts/overview/working-with-objects/common-labels/#labels).
 
+### kubectl.kubernetes.io/restartedAt
+
+Example: `kubectl.kubernetes.io/restartedAt: '2023-03-05T15:30:00Z`
+
+Used on: Pod
+
+This annotation is automatically added to the spec template by kubectl when the workload is restarted. It records the time of the most recent workload restart as a Unix timestamp.
```
Comment (NitishKumar06) [inline content/en/docs/reference/labels-annotations-taints/_index.md:101]:
```suggestion
The annotation indicates the time when a Kubernetes resource was last restarted using the kubectl command-line tool. It records the time of the most recent workload restart as a Unix timestamp.
```

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 4]
```diff
@@ -92,6 +92,18 @@ Common forms of values include:
...
+
+In the above example, the annotation indicates that the Pod was most recently restarted at 3:30 PM UTC on March 31, 2023.
+
+The `"kubectl.kubernetes.io/restartedAt"` annotation is used by Kubernetes to keep track of the timestamp of the most recent container restart for a particular pod. This information can be useful for debugging and monitoring purposes.
```
Comment (NitishKumar06) [inline content/en/docs/reference/labels-annotations-taints/_index.md:105]:
```suggestion
The `"kubectl.kubernetes.io/restartedAt"` annotation is used by Kubernetes to keep track of the timestamp of the most recent container restart for a particular pod. This information can be useful for troubleshooting and auditing purposes, allowing administrators to track when a resource was last restarted and correlate it with other events or changes in the cluster..
```

**Review by NitishKumar06 — COMMENTED** [review]
Ptal!

File: content/en/docs/reference/labels-annotations-taints/_index.md (hunk 2 above)
Comment (NitishKumar06) [reply content/en/docs/reference/labels-annotations-taints/_index.md]: what about this?

**Comment by sftim** [comment]
/retitle Add details for kubectl.kubernetes.io/restartedAt annotation

@Biswajit-Mohapatra2 do you know how to [squash commits](https://www.kubernetes.dev/docs/guide/pull-requests/#squashing) and to fix the invalid commit message?

**Comment by Biswajit-Mohapatra2** [comment]
> /retitle Add details for kubectl.kubernetes.io/restartedAt annotation
> 
> @Biswajit-Mohapatra2 do you know how to [squash commits](https://www.kubernetes.dev/docs/guide/pull-requests/#squashing) and to fix the invalid commit message?

please help me to do! @sftim

**Comment by NitishKumar06** [comment]
> please help me to do! @sftim

1. Make sure your local branch is in sync with remote master branch.
2. Copy the commit hash you'd like to include in a squash using `git log`
3. Once copied, `git rebase -i <commit-hash>`
4. Add a `pick` `squash` to the commit you'd like to take in consideration.
5. Save and Write a commit message for the final commit.
6. Exit and push the changes using `-f`.

**Comment by Ritikaa96** [comment]
For whoever was tracking/reviewing this PR ,
new PR is raised for the same.
#41348

**Pull request closed without merging**